src/common/modules/UnicodeFontHandler.js

```diff
@@ -154,75 +143,60 @@ function handle(info, tab) {
  * @returns {void}
  */
 function applySettings(unicodeFont) {
-	const menus = browser.menus || browser.contextMenus; // fallback for Thunderbird
-
-	menus.removeAll();
-
-	if (unicodeFont.changeCase) {
-		for (const id of caseIds) {
-			// .replaceAll(" ", "-");
-			const aid = id.toLowerCase().split(" ").join("-");
-			menus.create({
-				"id": aid,
-				"title": changeCase[aid](id),
-				"contexts": ["editable"],
-			});
-		}
-	}
-
-	if (unicodeFont.changeCase && unicodeFont.changeFont) {
-		menus.create({
-			id: "separator-2",
-			type: "separator",
-			contexts: ["editable"]
-		});
-	}
-
-	if (unicodeFont.changeFont) {
-		for (const id of fontIds) {
-			if (id === "separator") {
-				menus.create({
-					// id: id,
-					type: "separator",
-					contexts: ["editable"]
-				});
-			} else {
-				// .replaceAll(" ", "-");
-				const aid = id.toLowerCase().split(" ").join("-");
-				// console.log(id, aid, fonts[aid], changeFont(id, aid));
-				menus.create({
-					"id": aid,
-					"title": changeFont(id, aid),
-					"contexts": ["editable"],
-				});
-			}
-		}
-	}
+    const menus = browser.menus || browser.contextMenus; // fallback for Thunderbird
+
+    menus.removeAll();
+
+    let menuItemsToShow = contextMenuList;
+    if (!unicodeFont.changeCase) {
+        const caseValues = Object.values(CASE);
+        menuItemsToShow = contextMenuList.filter((id) => !caseValues.includes(id) );
+    }
+    if (!unicodeFont.changeFont) {
+        const fontValues = Object.values(FONT);
+        menuItemsToShow = contextMenuList.filter((id) => !fontValues.includes(id) );
+    }
+    menuItemsToShow
+        .filter((id, index) => (index !== 0 || id !== SEPARATOR))
+        .forEach((id) => {
+            if (id === SEPARATOR) {
+                menus.create({
+                    type: "separator",
+                    contexts: ["editable"]
+                });
+            } else {
+                menus.create({
+                    "id": id.toString(),
+                    "title": browser.i18n.getMessage(menuTranslation[id]),
```

Comment:
This line prevents the user from seeing the effect of the options.

Reply:
Yes, I do plan to move the effect of the translations into the translation itself. As said, to make things like `надстрочный индекс (ˢᵘᵖᵉʳˢᶜʳⁱᵖᵗ)` possible and allow translators to customize the appearance, if it needs to.

Reply:
OK, [as I said](https://github.com/rugk/unicodify/pull/5#issuecomment-736367408) it would be possible either way since the `changeFont()` function ignores non-ASCII characters...

Reply:
Okay okay, we'll try it like this.